stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/EventHandlerImpl.kt

```diff
@@ -65,65 +65,109 @@ import io.getstream.chat.android.offline.experimental.extensions.state
 import io.getstream.chat.android.offline.extensions.mergeReactions
 import io.getstream.chat.android.offline.extensions.setMember
 import io.getstream.chat.android.offline.extensions.updateReads
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 internal class EventHandlerImpl(
     private val domainImpl: ChatDomainImpl,
     private val client: ChatClient,
 ) {
+    private val eventBuffer = MutableSharedFlow<List<ChatEvent>>()
```

Comment:
`MutableSharedFlow` promises syncronization regardless of what thread the value was emitted from and its default strategy is to suspend when the consumer (collector) is busy. This way we do not lose any events.